covsirphy/cleaning/jhu_complement.py

```diff
@@ -197,10 +201,53 @@ def _monotonic(self, df, variable):
             series.fillna(method="ffill", inplace=True)
             # Reduce values to the previous date
             df.loc[:date, variable] = series * raw_last / series.iloc[-1]
-            df[variable] = df[variable].fillna(0).astype(np.int64)
+            df[variable] = df[variable].fillna(0)
+        df[variable] = df[variable].apply(np.ceil).astype(np.int64)
         self.complement_dict[f"Monotonic_{variable.lower()}"] = True
         return df
 
+    def _validate_recovery_period(self, df, upper_limit_days=90, lower_limit_days=7,
+                                  upper_percentage=0.5, lower_percentage=0.5):
+        """
+        Calculates and validates recovery period for specific country
+        as an additional condition in order to apply full complement or not
+
+        Args:
+            df (pandas.DataFrame):
+                - Index: reset_index
+                - Columns: Date, Confirmed, Recovered, Fatal
+            upper_limit_days (int): maximum number of valid partial recovery periods [days]
+            lower_limit_days (int): minimum number of valid partial recovery periods [days]
+            upper_percentage (float): fraction of partial recovery periods with value greater than upper_limit_days
+            lower_percentage (float): fraction of partial recovery periods with value less than lower_limit_days
+
+        Returns:
+            bool: true if recovery period is within valid range or false otherwise
+
+        Note: Passed argument df corresponds to specific country's df
+              upper_limit_days has default value of 3 months (90 days)
+              lower_limit_days has default value of 1 week (7 days)
+        """
+        recovery_period_is_valid = True
+        check_df = df.copy()
```

Comment:
I think it is unnecessary to copy it using memory, because we do not use the input dataframe later in this method.